crates/tabby/src/routes/chat.rs

```diff
@@ -31,17 +31,10 @@ pub async fn chat_completions(
     Json(request): Json<ChatCompletionRequest>,
 ) -> Response {
     let s = stream! {
-        for await content in state.generate(&request).await {
-            let content = match serde_json::to_string(&content) {
-                Ok(s) => s,
-                Err(e) => {
-                    yield Err(e);
-                    continue
-                }
-            };
-
+        for await content in state.generate(&request).await? {
```

Comment:
We should handle the error type in state.generate, with a match clause.

e.g for Minijinja error, we'd like to returns something like https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422

You'll also need to update the status code responses in line 22.